### ISSUE TITLE
rsync: Trailing slash of source removed

### ISSUE THREAD

**Issue by zxyrepf** [drakkan/sftpgo#327]
Hi,

rsync has different behavior for paths with a trailing slash:

> A trailing slash on the source changes this behavior to avoid creating an additional directory level at the destination. You can think of a trailing / on a source as meaning "copy the contents of this directory" as opposed to "copy the directory by name", but in both cases the attributes of the containing directory are transferred to the containing directory on the destination.
> https://man.archlinux.org/man/extra/rsync/rsync.1.en#USAGE

It looks like trailing slashes are removed by SFTPGo's internal path resolution before executing rsync. Would it be possible to keep them?

Thank you!

**Comment by drakkan** [comment]
Hi,

thanks for reporting this issue, I am finalizing another feature now, I will have a look in the next few days

**Comment by drakkan** [comment]
Hi,

I did a quick test. I have the user `rsync` with home dir `/tmp/rsync`, please take a look here:

```
ls testdir/
file.txt

rsync -avz -e "ssh -p 2022" testdir rsync@127.0.0.1:/dest
rsync@127.0.0.1's password: 
sending incremental file list
testdir/
testdir/file.txt

sent 1,497 bytes  received 39 bytes  614.40 bytes/sec
total size is 4,034  speedup is 2.63
```

and I have `testdir` inside `dest`:

```
ls /tmp/rsync/dest/
testdir
```

If I use this command:

```
sync -avz -e "ssh -p 2022" testdir/ rsync@127.0.0.1:/dest
rsync@127.0.0.1's password: 
sending incremental file list
./
file.txt

sent 1,480 bytes  received 38 bytes  433.71 bytes/sec
total size is 4,034  speedup is 2.66
```

I now have the `testdir` contents inside `dest`:

```
ls /tmp/rsync/dest/
file.txt
``` 

so it seems to work as expected. 

Please provide the command you use, the expected result and the actual result, thanks

**Comment by zxyrepf** [comment]
The issue occurs when SFTPGo is used as source:

```
$ rsync -avz -e "ssh -p 2022" rsync@127.0.0.1:/dest/testdir testdir
receiving incremental file list
created directory testdir
testdir/
testdir/file.txt

sent 47 bytes  received 138 bytes  370.00 bytes/sec
total size is 0  speedup is 0.00

$ rm -r testdir

$ rsync -avz -e "ssh -p 2022" rsync@127.0.0.1:/dest/testdir/ testdir
receiving incremental file list
created directory testdir
testdir/
testdir/file.txt

sent 47 bytes  received 138 bytes  370.00 bytes/sec
total size is 0  speedup is 0.00
```

The expected behavior would be the one observed in your examples.

It seems like the the trailing slash case is already handled in [`cleanCommandPath`](https://github.com/drakkan/sftpgo/blob/534b253c20e40707f677421a0bdc933f940445bb/sftpd/ssh_cmd.go#L528-L530), but [`ResolvePath`](https://github.com/drakkan/sftpgo/blob/534b253c20e40707f677421a0bdc933f940445bb/sftpd/ssh_cmd.go#L462) removes it again.

**Comment by drakkan** [comment]
Hi,

can you please confirm that it now works as expected? Thank you for your patience

**Comment by zxyrepf** [comment]
Yes, it works as expected now. Thank you for the quick fix and in general for the awesome software!

**Comment by drakkan** [comment]
Thanks for confirming, I have backported the change to the `2.0.x` branch as well so it will be available in v2.0.3 when I'll release it
